wolff/optimization/gradient_descent.py

```diff
@@ -0,0 +1,46 @@
+'''Gradient Descent Step'''
+from wolff.layer.layer import Layer
+
+
+class GradientDescent(Layer):
+
+    def __init__(self, theta=[], grad_J=[], learning_rate=0.001):
+        super(GradientDescent, self).__init__()
+
+        self.theta = theta
+        self.grad_J = grad_J
+        self.learning_rate = learning_rate
+
+    @property
+    def theta(self):
+        return self._theta
+
+    @theta.setter
+    def theta(self, value):
+        self._theta = value
```

Comment:
Since this is a list, we need a deep copy.
```suggestion
        self._theta = list(value)
```